### ISSUE TITLE
The table wizard should align the columns by default

### ISSUE THREAD

**Issue by dbitouze** [texstudio-org/texstudio#790]
Currently, the table wizard doesn't align the columns. In order to have columns aligned, the user has to apply LaTeX → Manipulate Tables → Align Columns. I don't see any benefit of not having columns aligned by default.

**Comment by foreachthing** [comment]
It's easier to add the alignment manually than removing it ... unless there will be an un-align button as well.
If you have long text in cells, the it gets really messy with auto-align.

**Comment by MeanSquaredError** [comment]
I am used to creating the tables manually but I looked at the Wizards drop-down and did not see a table wizard, but noticed "Quick tabular", which does most of the work when you need to create a table. Is this menu item what you call "table wizard" ?

In TXS 2.12.16 "Quick tabular" has a drop-down called "Alignment" which by default is set to "Center" and which centers the columns by setting the code to
`\begin{tabular}{|c|c|}`
Isn't that what you actually suggest?

**Comment by dbitouze** [comment]
> It's easier to add the alignment manually than removing it ... unless there will be an un-align button as well.

Makes sense.

> If you have long text in cells, the it gets really messy with auto-align.

Makes sense. Hence un-align button would be needed but maybe puzzling at a first glance.

**Comment by benibela** [comment]
>Makes sense. Hence un-align button would be needed but maybe puzzling at a first  glance.

Could be called table compression or table trimming

**Comment by dbitouze** [comment]
> I am used to creating the tables manually but I looked at the Wizards drop-down and did not see a table wizard, but noticed "Quick tabular", which does most of the work when you need to create a table. Is this menu item what you call "table wizard" ?

Indeed, sorry for the misnaming. I have been misled by the name of the LaTeX → Manipulate Tables menu entry. In order to be consistent, only one of "table" and "tabular" should be used in TXS. Though "tabular" seems to be more natural (the corresponding LaTeX environment is `tabular`), one could object that, in English, "tabular" seems to not be the right word and "table" seems to be the right one (caveat: English is not my mother tongue).

>In TXS 2.12.16 "Quick tabular" has a drop-down called "Alignment" which by default is set to "Center" and which centers the columns by setting the code to
`\begin{tabular}{|c|c|}`
Isn't that what you actually suggest?

No. What I suggest is to (silently) apply LaTeX → Manipulate Tables → Align Columns to the LaTeX code generated by the Wizards → Quick tabular. E.g. if you apply this wizard to generate the following table:

<table>
  <tr>
    <td>aaa</td>
    <td>bbb</td>
  </tr>
  <tr>
    <td>a</td>
    <td>b</td>
  </tr>
</table>

the resulting LaTeX code is currently:
```latex
\begin{tabular}{|c|c|}
	\hline
	aaa & bbb \\
	\hline
	a & b \\
	\hline
\end{tabular}
```
but, if LaTeX → Manipulate Tables → Align Columns is applied, becomes (note the vertically aligned `&`s):
```latex
\begin{tabular}{|c|c|}
	\hline
	aaa & bbb \\ \hline
	 a  &  b  \\ \hline
\end{tabular}
```

**Comment by dbitouze** [comment]
> Could be called table compression

Indeed. 

> or table trimming

Maybe I'm wrong but wouldn't it be confusing, as "trimming" may have specific meanings?

**Comment by MeanSquaredError** [comment]
So this LaTeX → Manipulate Tables → Align Columns is a kind of tool to format .tex source code?

**Comment by MeanSquaredError** [comment]
Thinking again about it if it is implemented as an optional checkbox then maybe it is possible to remember the last wizard settings including this checkbox state so that those who prefer formatted source code will have it enabled by default.

**Comment by dbitouze** [comment]
> > Makes sense. Hence un-align button would be needed but maybe puzzling at a first  glance.
> 
> Could be called table compression or table trimming

@benibela Another option: the LaTeX → Manipulate Tables → Align Columns (and corresponding button) could be a toggle between "Columns Aligned" and "Columns Compressed".

**Comment by dbitouze** [comment]
> So this LaTeX → Manipulate Tables → Align Columns is a kind of tool to format .tex source code?

Indeed.

> If the purpose of this function is just to format the source code to make it look nicer, then I see at least one reason to NOT enable this kind of formatting by default. In my case I have tables with many columns. [...] At this point I would prefer having the row not formatted (i.e. without any extra spaces) because it allows one table row to take fewer display rows.

Makes sense.

> So IMHO this source code formatting feature should be made optional and (preferably) disabled by default. However I see the benefit from it to some people so maybe it should be activated by a checkbox in the tabular wizard. The checkbox should be off by default.
>
> Thinking again about it if it is implemented as an optional checkbox then maybe it is possible to remember the last wizard settings including this checkbox state so that those who prefer formatted source code will have it enabled by default.

Could be nice but, as suggested by @foreachthing, being able to toggle between the two kinds of  tables' code formatting would be nice, and even necessary (if the user forgets to change his last wizard setting that isn't the best one for the table he just created).
